### PR TITLE
toggle comment shortcut

### PR DESCRIPTION
**Pull request by reisraff**

### CODE REVIEW

File: editor/code_editor.cpp [hunk 1]
```diff
@@ -634,6 +634,9 @@ void CodeTextEditor::_text_editor_gui_input(const Ref<InputEvent> &p_event) {
 			if (ED_IS_SHORTCUT("script_editor/reset_zoom", p_event)) {
 				_reset_zoom();
 			}
+			if (ED_IS_SHORTCUT("script_editor/toogle_comment", p_event)) {
+				_toogle_comment();
+			}
```
Comment (LikeLakers2) [inline editor/code_editor.cpp]: `toggle`, not `toogle`

**Comment by LeonardMeagher2** [comment]
Doesn't Ctrl+K already do this?

**Comment by LeonardMeagher2** [comment]
https://github.com/godotengine/godot/blob/07b76c0376eef34fcbf52ecc21074c5084f980f5/editor/plugins/script_text_editor.cpp#L1515

I guess ScriptTextEditor is different than the CodeEditor, but I imagine the default shortcuts should be the same. I know Ctrl+/ is pretty standard, I wonder why Ctrl+K is used in the script editor

Sorry for the off topic comment

File: scene/gui/text_edit.cpp [hunk 2]
```diff
@@ -3195,6 +3201,34 @@ void TextEdit::_uncomment_line(int p_line) {
...
+
+int TextEdit::get_line_indent_size(int p_line) const {
+	String line_text = get_line(p_line);
+
+	if (line_text[0] == '\n') {
+		return -1;
+	}
+
+	for (int i = 0; i < line_text.length(); i++) {
+		if (line_text[i] == '\t') {
```
Comment (Paulb23) [inline scene/gui/text_edit.cpp]: Need to check for spaces as well, there's a helper method, `_is_whitespace()`.

**Review by Paulb23 — CHANGES_REQUESTED** [review]
I assume this is designed to fix #22177?

Rather then a new method in `CodeTextEditor` and shortcut.  it should be possible to change the existing method `toggle_inline_comment` (`CTRL+K`). 

In addition the comment / comment methods shouldn't be in `TextEdit`, but `CodeTextEditor`.

**Comment by akien-mga** [comment]
I'm not sure what this does which is not already supported with the Ctrl+K toggle comment hotkey?

@Paulb23's comment needs to be addressed in any case, and the commits should be squashed together with a clearer commit log for the final commit.

**Comment by akien-mga** [comment]
Closing as per the above, the feature already exists via `toggle_inline_comment` so it should be expanded to support what is needed in #22177.

**Pull request closed without merging**